leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -84,6 +87,10 @@ private AwaitingLeadershipProxy(Supplier<?> delegateSupplier,
         this.delegateRef = new AtomicReference<Object>();
         this.interfaceClass = interfaceClass;
         this.isClosed = false;
+
+        String logTag = "Proxy #" + numProxies.getAndIncrement() + " - " + interfaceClass.getSimpleName();
+        log = new TaggedLogger(AwaitingLeadershipProxy.class, logTag);
+        leaderLog = new TaggedLogger("leadership", logTag);
```

Comment:
Could consider just making tags part of logger name so that it would be something like `com.palantir....AwaitingLeadershipProxy.TimestampService.N` then one could handle sending different interfaces to different appenders, and there is no need for the wrapper or string concatenation at logging time on our side.